### PR TITLE
Changed FormField to support background styling tied to error and disabled states

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed FormField to support background styling tied to error and disabled states

#### Where should the reviewer start?

FormField/doc.js

#### What testing has been done on this PR?

storybook, theme designer

#### How should this be manually tested?

storybook

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/themes/base.js [hunk 1]
```diff
@@ -517,17 +517,18 @@ export const generate = (baseSpacing = 24, scale = 6) => {
...
       },
       content: {
-        pad: {
-          horizontal: 'small',
-          bottom: 'small',
-        },
+        pad: 'small',
+      },
+      disabled: {
+        // background: undefined,
```
Comment (ShimiSun) [inline src/js/themes/base.js]: `base.d.ts` should include 
```
 disabled?: {
    background?: BackgroundType,
}
```
here https://github.com/grommet/grommet/blob/c4f9d42dc862bddde612172237a57aa7003c648a/src/js/themes/base.d.ts#L475

File: src/js/themes/base.js [hunk 2]
```diff
@@ -517,17 +517,18 @@ export const generate = (baseSpacing = 24, scale = 6) => {
...
       },
       error: {
         color: {
           dark: 'status-critical',
           light: 'status-critical',
         },
         margin: { vertical: 'xsmall', horizontal: 'small' },
+        // background: undefined,
```
Comment (ShimiSun) [inline src/js/themes/base.js]: Should be added in a similar way to https://github.com/grommet/grommet/blob/c4f9d42dc862bddde612172237a57aa7003c648a/src/js/themes/base.d.ts#L478

File: src/js/components/FormField/doc.js [hunk 3]
```diff
@@ -101,7 +104,29 @@ export const themeDoc = {
...
+    defaultValue: 'small',
+  },
+  'formField.disabled.background.color': {
+    description: 'The color of the FormField background when it is disabled.',
+    type: "string | {'dark': string, 'light': string}",
+    defaultValue: undefined,
+  },
+  'formField.disabled.background.opacity': {
+    description: 'The opacity of the FormField background when it is disabled.',
+    type: 'string',
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js]: According to https://github.com/grommet/grommet/blob/c4f9d42dc862bddde612172237a57aa7003c648a/src/js/utils/background.js#L81

I think that `background.opacity` could also be boolean and a number.

File: src/js/themes/base.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/themes/base.js]: Ah, yes. I forgot typescript for base.js. Good catch!

File: src/js/themes/base.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/themes/base.js]: yep

**Pull request merged**